### PR TITLE
Check app clip

### PR DESCRIPTION
**Pull request by gre4ixin**

### CODE REVIEW

File: Sources/AsyncLocationKit/AsyncLocationManager.swift [hunk 1]
```diff
@@ -45,6 +45,14 @@ public final class AsyncLocationManager {
         locationManager.desiredAccuracy = desiredAccuracy.convertingAccuracy
     }
     
+    public init(locationManager: CLLocationManager, desiredAccuracy: LocationAccuracy) {
+        self.locationManager = locationManager
+        self.locationManager.delegate = locationDelegate
```
Comment (appfrilans) [inline Sources/AsyncLocationKit/AsyncLocationManager.swift:50]: I get the error "Variable 'self.locationDelegate' used before being initialized" on this line

Comment (gre4ixin) [reply Sources/AsyncLocationKit/AsyncLocationManager.swift:50]: sorry for this, this init was created for tests, I did not notice an error when pushing, corrected :)

Comment (appfrilans) [reply Sources/AsyncLocationKit/AsyncLocationManager.swift:50]: Thanks! Now it works just fine. And the solution to add the APPCLIP condition seems to work fine as well!

Comment (gre4ixin) [reply Sources/AsyncLocationKit/AsyncLocationManager.swift:50]: Nice 😊

Comment (appfrilans) [reply Sources/AsyncLocationKit/AsyncLocationManager.swift:50]: No, I celebrated to early. Uploading to TestFlight now gives the same error as before. I guess the APPCLIP flag does not propagate down to the SPM frameworks :( Not sure how to solve it now. I guess I either have to clone and copy down the code for this framework and the FLAG will be respected, or just skip the framework for my AppClip target and loose all GPS functionality for my AppClip-target :(

Comment (gre4ixin) [reply Sources/AsyncLocationKit/AsyncLocationManager.swift:50]: Can you use cocoapods?

Comment (appfrilans) [reply Sources/AsyncLocationKit/AsyncLocationManager.swift:50]: I've given up on Cocoapods a long time ago. SPM only now.

Comment (gre4ixin) [reply Sources/AsyncLocationKit/AsyncLocationManager.swift:50]: I'll think about how to solve your problem)

**Comment by tcobbs-bentley** [comment]
@gre4ixin : This change leads to compile-time warnings for anyone using the library, since `requestPermission(with:)` itself calls the now deprecated `requestAuthorizationAlways()` and `requestAuthorizationWhenInUse()`:

```
/<...>/SourcePackages/checkouts/AsyncLocationKit/Sources/AsyncLocationKit/AsyncLocationManager.swift:118:26: warning: 'requestAuthorizationAlways()' is deprecated: Use new function requestPermission(with:)
            return await requestAuthorizationAlways()
                         ^
/<...>/SourcePackages/checkouts/AsyncLocationKit/Sources/AsyncLocationKit/AsyncLocationManager.swift:121:26: warning: 'requestAuthorizationWhenInUse()' is deprecated: Use new function requestPermission(with:)
            return await requestAuthorizationWhenInUse()
                         ^
```

**Comment by gre4ixin** [comment]
@tcobbs-bentley use 1.5.6 version)

**Pull request merged**